logger_test.go

```diff
@@ -56,7 +56,45 @@ func TestPrivateBackend(t *testing.T) {
 	if stdBackend.size > 0 {
 		t.Errorf("something in stdBackend, size of backend: %d", stdBackend.size)
 	}
-	if "to private baсkend" == MemoryRecordN(privateBackend, 0).Formatted(0) {
-		t.Error("logged to defaultBackend:", MemoryRecordN(privateBackend, 0))
+	if privateBackend.size != 1 {
+		t.Errorf("privateBackend must contain something, size of backend: %d", privateBackend.size)
+	}
+	if "to private backend" != MemoryRecordN(privateBackend, 0).Formatted(0) {
+		t.Error("must be logged to privateBackend:", MemoryRecordN(privateBackend, 0))
+	}
+}
+
+func testConcurrent_Log(i int, sync *syncTestConcurrent, lvlBackend *LeveledBackend, log *Logger) {
+	sync.start.Done()
+	sync.start.Wait()
+	for j := 0; j < 1000; j++ {
+		log.SetBackend(*lvlBackend)
+		log.Debug("to private backend")
+	}
+	sync.end.Done()
+}
+
+func TestPrivateBackend_Concurency(t *testing.T) {
+	stdBackend := InitForTesting(DEBUG)
+	log := MustGetLogger("test")
+	privateBackend := NewMemoryBackend(10240)
+	lvlBackend := AddModuleLevel(privateBackend)
+
+	sync := &syncTestConcurrent{}
+	sync.end.Add(10)
+	sync.start.Add(10)
+	for i := 0; i < 10; i++ {
+		go testConcurrent_Log(i, sync, &lvlBackend, log)
```

Comment:
Concurrent use of `GetLevel` with `SetLevel` is not under test here I think. Thus why the `go test -race` probably isn't picking up the concurrent read/write access to the `levels` map